cwr/index_test.go

```diff
@@ -48,194 +50,198 @@ func (t *testIndex) cleanup() {
+	var (
+		objectID           string
+		title              string
+		iswc               string
+		publisherSequenceN string
+	)
+
+	rows, err := x.db.Query(`SELECT object_id,title,iswc FROM registered_work WHERE cwr_id = ?`, x.cwrCid.String())
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer x.cleanup()
-	// check all the publisherControlledBySubmitter were indexed
-	for _, record := range x.records {
-		// check the publisher_sequence_number indexes
-		if record.RecordType != "SPU" {
-			continue
+	defer rows.Close()
+
+	rowsCount := 0
+	for rows.Next() {
+		rowsCount++
+		if err = rows.Scan(&objectID, &title, &iswc); err != nil {
+			t.Fatal(err)
+		}
+		if err = check(objectID, map[string]string{
+			"title": title,
+			"iswc":  iswc,
+		}); err != nil {
+			t.Fatal(err)
 		}
-		rows, err := x.db.Query(
-			`SELECT object_id FROM publisher_control WHERE publisher_sequence_n = ? AND record_type = ?`,
-			record.PublisherSequenceNumber, record.RecordType,
-		)
+		//get all publisher control records (SPU) which link to the above tx .
+		txID, err := cid.Decode(objectID)
 		if err != nil {
 			t.Fatal(err)
 		}
-		defer rows.Close()
-		var objectID string
-		for rows.Next() {
-			// if we've already set objectID then we have a duplicate
-			if objectID != "" {
-				t.Fatalf("duplicate entries for SPU publisher sequence number %q", record.PublisherSequenceNumber)
-			}
-			if err := rows.Scan(&objectID); err != nil {
-				t.Fatal(err)
-			}
+		spuRows, err := x.db.Query(`SELECT object_id,publisher_sequence_n FROM publisher_control WHERE tx_id = ?`, txID.String())
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer spuRows.Close()
+		for spuRows.Next() {
 
-			// check we can get the object from the store
-			cid, err := cid.Parse(objectID)
-			if err != nil {
+			if err := spuRows.Scan(&objectID, &publisherSequenceN); err != nil {
 				t.Fatal(err)
 			}
-			obj, err := x.store.Get(cid)
-			if err != nil {
+			if err = check(objectID, map[string]string{
+				"publisher_sequence_n": publisherSequenceN,
+			}); err != nil {
 				t.Fatal(err)
 			}
-			// check the object has the correct fields
-			for key, expected := range map[string]string{
-				"publisher_sequence_n": record.PublisherSequenceNumber,
-				"record_type":          record.RecordType,
-			} {
-				actual, err := obj.GetString(key)
-				if err != nil {
-					t.Fatal(err)
-				}
-				if actual != expected {
-					t.Fatalf("expected object %s to be %q, got %q", key, expected, actual)
-				}
-			}
 		}
+	}
+	if rowsCount == 0 {
+		t.Fatal("no registered_work found")
+	}
+}
+func openStore(metaDir string) (*meta.Store, error) {
 
-		// check we got an object and no db errors
-		if objectID == "" {
-			t.Fatalf("SPU %q not found", record.PublisherSequenceNumber)
-		} else if err := rows.Err(); err != nil {
-			t.Fatal(err)
+	store, err := fs.NewDatastore(metaDir)
+	if err != nil {
+		return nil, err
+	}
+	return meta.NewStore(store), nil
```

Comment:
Perhaps this should be moved to `meta.NewFSStore(dir)` and used both here and in the CLI.

Reply:
Can do that...though the swarm datastore is nocking on the door..should we ?

Reply:
We're still going to have the file store, I think it is worth it.